src/local/room_xy.rs

```diff
@@ -229,6 +246,19 @@ impl RoomXY {
     }
 }
 
+impl PartialOrd for RoomXY {
+    #[inline]
+    fn partial_cmp(&self, other: &RoomXY) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for RoomXY {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.y.cmp(&other.y).then_with(|| self.x.cmp(&other.x))
```

Comment:
Nit; likely less efficient than
```suggestion
        (self.y, self.x).cmp(&(other.y, other.x))
```

Reply:
Learn something new every day. I wasn't aware of the default implementation of short-circuiting comparisons for tuples. I agree that it's likely more efficient.